### PR TITLE
[VL] Correct task-wise off-heap memory calculation

### PR DESCRIPTION
**Pull request by zhztheplayer**

### CODE REVIEW

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

**Comment by zhztheplayer** [comment]
@zhouyuan @jinchengchenghh @rui-mo

**Review by zhouyuan — APPROVED** [review]
👍

**Review by jinchengchenghh — APPROVED** [review]

File: gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala [hunk 1]
```diff
@@ -55,12 +55,27 @@ private[glutenproject] class GlutenDriverPlugin extends DriverPlugin {
...
+    if (!conf.contains(GlutenConfig.GLUTEN_OFFHEAP_SIZE_KEY)) {
+      throw new UnsupportedOperationException(s"${GlutenConfig.GLUTEN_OFFHEAP_SIZE_KEY} is not set")
+    }
+    val offHeapSize = conf.getSizeAsBytes(GlutenConfig.GLUTEN_OFFHEAP_SIZE_KEY)
+    conf.set(GlutenConfig.GLUTEN_OFFHEAP_SIZE_IN_BYTES_KEY, offHeapSize.toString)
+    // FIXME Is this calculation always reliable ? E.g. if dynamic allocation is enabled
+    val executorCores = conf.getInt("spark.executor.cores", 1)
+    val taskCores = conf.getInt("spark.task.cpus", 1)
+    val offHeapPerTask = offHeapSize / (executorCores / taskCores)
+    conf.set(GlutenConfig.GLUTEN_TASK_OFFHEAP_SIZE_IN_BYTES_KEY, offHeapPerTask.toString)
```
Comment (jinchengchenghh) [inline gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala:76]: What if user set `GLUTEN_TASK_OFFHEAP_SIZE_IN_BYTES_KEY`

**Pull request merged**